### PR TITLE
Change the import of kenlm from github to pypi

### LINKED ISSUE #19686: Make use of KenLM's pypi version
### Feature request

Use PyPi version of KenLM instead of installing archieve file from Github.

### Motivation

Now that kenlm is on pypi: https://github.com/kpu/kenlm/issues/363#issuecomment-1280050903 let's use it :-) 

### Your contribution

@sanchit-gandhi would you give this a try as an exercise to see how we deal with soft dependencies in Transformers maybe? :-) 

I think we can now update the `setup.py` file to include the pypi version of `kenlm` and remove all those `pip install https://github.com/kpu/kenlm/archive/master.zip` statements  e.g. in https://github.com/huggingface/transformers/blob/c7edde1a692012eda23bc2b837588557b97ad729/docker/transformers-all-latest-gpu/Dockerfile#L44

I'd also advocate to add a `is_kenlm_available` flag to give a nice error message and update warnings / errors here: https://github.com/huggingface/transformers/blob/c7edde1a692012eda23bc2b837588557b97ad729/src/transformers/pipelines/__init__.py#L839

### PR DESCRIPTION
**Pull request by raghavanone**
# What does this PR do?

Fixes #19686 

## Who can review?

@patrickvonplaten

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
Thanks for your PR. The main interest of having the package on PiPy is that we can put it in our setup and not have to write a specific command to install it. You should add it to the extras for audio.

**Comment by raghavanone** [comment]
> Thanks for your PR. The main interest of having the package on PiPy is that we can put it in our setup and not have to write a specific command to install it. You should add it to the extras for audio.

Done

File: setup.py [hunk 1]
```diff
@@ -123,6 +123,7 @@
     "jax>=0.2.8,!=0.3.2,<=0.3.6",
     "jaxlib>=0.1.65,<=0.3.6",
     "jieba",
+    "kenlm==0.1",
```
Comment (sgugger) [inline setup.py:126]: Why the specific pin, we want to test updated versions.

File: docker/transformers-all-latest-gpu/Dockerfile [hunk 2]
```diff
@@ -41,7 +41,7 @@ RUN python3 -m pip uninstall -y flax jax
 RUN python3 -m pip install --no-cache-dir torch-scatter -f https://data.pyg.org/whl/torch-$(python3 -c "from torch import version; print(version.__version__.split('+')[0])")+$CUDA.html
 RUN python3 -m pip install --no-cache-dir intel_extension_for_pytorch==$INTEL_TORCH_EXT+cpu -f https://software.intel.com/ipex-whl-stable
 
-RUN python3 -m pip install --no-cache-dir git+https://github.com/facebookresearch/detectron2.git pytesseract https://github.com/kpu/kenlm/archive/master.zip
+RUN python3 -m pip install --no-cache-dir "kenlm==0.1"
```
Comment (sgugger) [inline docker/transformers-all-latest-gpu/Dockerfile:44]: Should be removed

File: .circleci/create_circleci_config.py [hunk 3]
```diff
@@ -127,7 +127,7 @@ def job_name(self):
         "pip install .[sklearn,tf-cpu,torch,testing,sentencepiece,torch-speech,vision]",
         TORCH_SCATTER_INSTALL,
         "pip install tensorflow_probability",
-        "pip install https://github.com/kpu/kenlm/archive/master.zip",
+        "pip install kenlm==0.1",
```
Comment (sgugger) [inline .circleci/create_circleci_config.py:130]: All of those can be removed now.

**Review by sgugger — APPROVED** [review]
Yay, way better, thanks! @ydshieh could you also have a look?

**Comment by ydshieh** [comment]
I think we still have 2 files to be clean up ?

- docker/transformers-pytorch-gpu/Dockerfile
- docker/transformers-doc-builder/Dockerfile

File: docker/transformers-all-latest-gpu/Dockerfile [hunk 4]
```diff
@@ -41,7 +41,6 @@ RUN python3 -m pip uninstall -y flax jax
 RUN python3 -m pip install --no-cache-dir torch-scatter -f https://data.pyg.org/whl/torch-$(python3 -c "from torch import version; print(version.__version__.split('+')[0])")+$CUDA.html
 RUN python3 -m pip install --no-cache-dir intel_extension_for_pytorch==$INTEL_TORCH_EXT+cpu -f https://software.intel.com/ipex-whl-stable
 
-RUN python3 -m pip install --no-cache-dir git+https://github.com/facebookresearch/detectron2.git pytesseract https://github.com/kpu/kenlm/archive/master.zip
```
Comment (ydshieh) [inline docker/transformers-all-latest-gpu/Dockerfile:44]: We should not remove the whole line, just the part `https://github.com/kpu/kenlm/archive/master.zip`

**Review by ydshieh — CHANGES_REQUESTED** [review]
See the comment, but we are close to merge. Thank you!

File: docker/transformers-all-latest-gpu/Dockerfile (hunk 4 above)
Comment (ydshieh) [reply docker/transformers-all-latest-gpu/Dockerfile:44]: Hi @raghavanone , once this comment is addressed, we are good to merge I think :-)

**Review by ydshieh — APPROVED** [review]
Very nice @raghavanone ! Thank you for taking care of this `kenlm` dependency 💯

**Pull request merged**